src/plugins/appeals/unbanThread.ts

```diff
@@ -0,0 +1,35 @@
+import {useEvent} from "../../hooks";
+import {Interaction, Message} from "discord.js";
+
+useEvent("interactionCreate", async (interaction: Interaction) => {
+    if (!interaction.isButton()) {
+        return;
+    }
+    const {customId} = interaction;
+    const [interactionType, action, userTag, userId] = customId.split("-");
+    const {message} = interaction;
+    if (interactionType !== "appeal") {
+        return;
+    }
+    if (action !== "thread") {
+        return;
+    }
+    try {
+        await interaction.message.startThread ({
+            name: `Appeal Thread - ${userTag}`
+        })
+    } catch {
+        await interaction.reply({
+            content: `A thread was already created for <@${userId}>.`,
+            ephemeral: true,
+        });
+        return;
+    }
+    if (message instanceof Message) {
+        await interaction.reply({
+            // Replace interaction.user with appealee's username.
+            content: `The thread for <@${userId}> is created.`,
```

Comment:
It is also common practice across g'day to add the user's username in parentheses immediately following their mention as Discord will not actually parse the mention if the mentioned user's profile is not in the client's cache. Including the username as well decreases this potential ambiguity. For instance, `The thread for ${userMention(userId)} (${username}) is created.`

Reply:
Username to me sounds like the user’s display name, which I’m not sure if that’s possible. If there’s a better solution, feel free to reply, I went with `${userMention(userId)} (${userId})`, like the User field in appeal entries.